platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java

```diff
@@ -0,0 +1,145 @@
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class used to migrate {@link ConfigurationAdmin} configurations. This includes importing
+ * configuration files from a configuration directory and creating {@link Configuration} objects
+ * for those and exporting {@link Configuration} objects to configuration files.
+ */
+public class ConfigurationAdminMigratable implements Migratable {
+    /**
+     * Holds the current export version.
+     * <p>
+     * 1.0 - initial version
+     */
+    private static final String VERSION = "1.0";
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(ConfigurationAdminMigratable.class);
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    private final PersistenceStrategy defaultStrategy;
+
+    public ConfigurationAdminMigratable(ConfigurationAdmin configurationAdmin,
+            String defaultFileExtension) {
+        Validate.notNull(configurationAdmin, "invalid null config admin");
+        Validate.notNull(defaultFileExtension, "invalid null default file extension");
+        this.configurationAdmin = configurationAdmin;
+        this.defaultStrategy = getPersister(defaultFileExtension);
+        Validate.notNull(defaultStrategy,
+                "unknown persistence strategy extension: " + defaultFileExtension);
+    }
+
+    @Override
+    public String getVersion() {
+        return ConfigurationAdminMigratable.VERSION;
+    }
+
+    @Override
+    public String getId() {
+        return "config.admin";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Configuration Admin Migration";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Exports Configuration Admin";
+    }
+
+    @Override
+    public String getOrganization() {
+        return "Codice";
+    }
+
+    @Override
+    public void doExport(ExportMigrationContext context) {
+        final ExportMigrationConfigurationAdminContext adminContext =
+                new ExportMigrationConfigurationAdminContext(context,
+                        this,
+                        getConfigurations(context));
+
+        adminContext.entries()
+                .forEach(ExportMigrationEntry::store);
+    }
+
+    @Override
+    public void doImport(ImportMigrationContext context) {
+        final ImportMigrationConfigurationAdminContext adminContext =
+                new ImportMigrationConfigurationAdminContext(context,
+                        this,
+                        configurationAdmin,
+                        getConfigurations(context));
+
+        adminContext.entries()
+                .forEach(ImportMigrationEntry::restore);
+    }
+
+    public PersistenceStrategy getDefaultPersister() {
+        return defaultStrategy;
+    }
+
+    public PersistenceStrategy getPersister(String extension) {
+        // TODO: change to dynamically lookup strategies
```

Comment:
❓ Do we still need to address this TODO as part of this PR? If not, we should remove it.

Reply:
It will be addressed in another ticket. I guess I left it there to make sure we don't forget. I'll remove since we do have the other ticket on the board.